lib/ansible/plugins/action/__init__.py

```diff
@@ -569,6 +569,9 @@ def _remote_expand_user(self, path, sudoable=True, pathsep=None):
 
         if sudoable and expand_path == '~' and self._play_context.become and self._play_context.become_user:
             expand_path = '~%s' % self._play_context.become_user
+        else:
+            # use remote user instead, if none set default to current user
+            expand_path = '~%s' % self._play_context.remote_user or self._connection.default_user or ''
```

Comment:
I think that this isn't doing what you expect, the code does

`expand_path = ('~%s' % self._play_context.remote_user) or self._connection.default_user or ''`

instead of

`expand_path = '~%s' % (self._play_context.remote_user or self._connection.default_user or '')`

Reply:
Thanks. It's already fixed in `devel` ;-)